include/Introspection.h

```diff
@@ -0,0 +1,1501 @@
+    /// in, but no enforcement is made that that happens.  If any else is passed in something will break.
+    /// @param ToMerge A list of MemberAccessor tuples to be bundled together.
+    /// @return Returns a Tuple containing all the tuples merged into one bigger tuple without duplicate types.
+    template<typename... TupleTypes>
+    constexpr auto MergeMembersUnique(TupleTypes&&... ToMerge)
+    {
+        constexpr auto Comp = [](const auto Left, const auto Right) constexpr {
+            return Left.GetName() == Right.GetName();
+        };
+        return tuple_cat_unique_comp(Comp,std::forward<TupleTypes>(ToMerge)...);
+    }
+
+    /// @brief Checks to see if a class type has members registered.
+    /// @tparam Class The class type to check.
+    /// @return Returns true if introspection is possible for members of the given class, false otherwise.
+    template<typename Class>
+    constexpr Boole IsRegistered()
+    {
+        using TupleType = decltype(IntrospectionHelpers::GetMembers<Class>());
+        return !std::is_same_v<std::tuple<>,TupleType>;
+    }
+
+    /// @brief Gets the registered name of the class type.
+    /// @tparam Class The class to check.
+    /// @return Returns the registered name of the class, or an empty String if no name is registered.
+    template<typename Class>
+    constexpr StringView GetRegisteredName()
+    {
+        return IntrospectionHelpers::GetName<Class>();
+    }
+
+    /// @brief Gets the number of registered members for a class.
+    /// @tparam Class The class to check.
+    /// @return Returns the amount of members that have been registered for the class specified.
+    template<typename Class>
+    constexpr size_t GetMemberCount()
+    {
+        using TupleType = decltype(IntrospectionHelpers::GetMembers<Class>());
+        return std::tuple_size_v<TupleType>;
+    }
+
+    /// @brief Gets a container of all the member accessors registered for a class.
+    /// @tparam Class The class to retrieve accessors for.
+    /// @return Returns a const reference to a tuple of member accessor types for each registered member.
+    template<typename Class>
+    const auto& GetRegisteredMembers()
+    {
+        using TupleType = decltype(IntrospectionHelpers::GetMembers<Class>());
+        return IntrospectionHelpers::MemberHolder<Class,TupleType>::Members;
+    }
+
+    /// @brief Invokes a callable for each registered member on a class.
+    /// @warning SFINAE will disable this function if the class is not registered.
+    /// @tparam Class Provided type who's members will be looped over.
+    /// @tparam Funct Deduced parameter for the callback method to invoke with each member.
+    /// @param ToCall The callback that will be invoked for each member in the specified class.
+    template<typename Class, typename Funct, typename = std::enable_if_t<IsRegistered<Class>()>>
+    void DoForAllMembers(Funct&& ToCall)
```

Comment:
I don't think it would make sense, no.  As we discussed before I think it would virtually never actually manage to be constexpr because of the function being passed in.  While we could mark it as constexpr for the less than 1% of the time where it might work, I think doing so would give users the wrong impression about that part of the API, trying to make promises that it can't reasonably keep.